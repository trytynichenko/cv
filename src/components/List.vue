<template>
  <div class="list">
    <div v-for="(category, i) in data" :key="i" class="list__category">
      <div class="list__category__label">{{ category.label }}</div>
      <ul class="list__category__item">
        <li v-for="(item, k) in category.items" :key="k">{{ item.name }}</li>
      </ul>
    </div>
  </div>
</template>

<script>
export default {
  name: 'List',
  props: {
    data: Array,
  },
};
</script>

<style scoped lang="scss">
  @import '../assets/styles/components/list';
</style>
