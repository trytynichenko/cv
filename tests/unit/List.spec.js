import { shallow } from '@vue/test-utils';
import List from '@/components/List.vue';

describe('List.vue', () => {
  const props = {
    data: [
      {
        label: 'label 1',
        items: [
          { name: 'item 1' },
          { name: 'item 2' },
          { name: 'item 3' },
        ],
      },
    ],
  };

  const wrapper = shallow(List, {
    propsData: props,
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
