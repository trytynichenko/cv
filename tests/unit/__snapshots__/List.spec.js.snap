// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue matches snapshot 1`] = `
<div class="list">
  <div class="list__category">
    <div class="list__category__label">label 1</div>
    <ul class="list__category__item">
      <li>item 1</li>
      <li>item 2</li>
      <li>item 3</li>
    </ul>
  </div>
</div>
`;
